core/src/main/scala/com/microsoft/azure/synapse/ml/causal/LinearDMLEstimator.scala

```diff
@@ -0,0 +1,279 @@
+import org.apache.spark.ml.{ComplexParamsReadable, ComplexParamsWritable, Estimator, Model, Pipeline}
+import org.apache.spark.ml.classification.ProbabilisticClassifier
+import org.apache.spark.ml.regression.{GeneralizedLinearRegression, Regressor}
+import org.apache.spark.ml.feature.VectorAssembler
+import org.apache.spark.ml.param.{DoubleArrayParam, ParamMap}
+import org.apache.spark.ml.param.shared.HasWeightCol
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.sql.{DataFrame, Dataset}
+import org.apache.spark.sql.types.StructType
+import scala.concurrent.Future
+
+/** Linear Double ML estimators. The estimator follows the two stage process,
+ *  where a set of nuisance functions are estimated in the first stage in a crossfitting manner
+ *  and a final stage estimates the average treatment effect (ATE) model.
+ *  Our goal is to estimate the constant marginal ATE Theta(X)
+ *
+ *  In this estimator, the ATE is estimated by using the following estimating equations:
+ *  .. math ::
+ *      Y - \\E[Y | X, W] = \\Theta(X) \\cdot (T - \\E[T | X, W]) + \\epsilon
+ *
+ *  Thus if we estimate the nuisance functions :math:`q(X, W) = \\E[Y | X, W]` and
+ *  :math:`f(X, W)=\\E[T | X, W]` in the first stage, we can estimate the final stage ate for each
+ *  treatment t, by running a regression, minimizing the residual on residual square loss,
+ *  estimating Theta(X) is a final regression problem, regressing tilde{Y} on X and tilde{T})
+ *
+ *  .. math ::
+ *       \\hat{\\theta} = \\arg\\min_{\\Theta}\
+ *       \E_n\\left[ (\\tilde{Y} - \\Theta(X) \\cdot \\tilde{T})^2 \\right]
+ *
+ * Where
+ * `\\tilde{Y}=Y - \\E[Y | X, W]` and :math:`\\tilde{T}=T-\\E[T | X, W]` denotes the
+ * residual outcome and residual treatment.
+ *
+ * The nuisance function :math:`q` is a simple regression problem and user
+ * can use setOutcomeModel to set an arbitrary sparkml regressor that is internally used to solve this regression problem
+ *
+ * The problem of estimating the nuisance function :math:`f` is also a regression problem and user
+ * can use setTreatmentModel to set an arbitrary sparkml regressor that is internally used to solve this regression problem.
+ *
+ * The input categorical treatment is one-hot encoded (excluding the lexicographically smallest treatment which is used as the baseline)
+ * and the `predict_proba` method of the treatment model classifier is used to residualize the one-hot encoded treatment.
+
+      The final stage is (potentially multi-task) linear regression problem with outcomes the labels
+      :math:`\\tilde{Y}` and regressors the composite features
+      :math:`\\tilde{T}\\otimes \\phi(X) = \\mathtt{vec}(\\tilde{T}\\cdot \\phi(X)^T)`.
+      The :class:`.DML` takes as input parameter
+      ``model_final``, which is any linear sparkml regressor that is internally used to solve this
+      (multi-task) linear regression problem.
+ */
+//noinspection ScalaStyle
+class LinearDMLEstimator(override val uid: String)
```

Comment:
If you havbe any plans to generalize this to nonlinear scenarios, perhaps we can call this DoubleMLEstimator and list "linear" in the docstring as a current limitation

Reply:
LinearDMLEstimator means that it run a linear regression for the last stage.  
@memoryz, do we think if we'll have nonlinear scenarios?

Reply:
Let's get some input from EconML team first.

Reply:
confirmed it from EconML team, we 'll go for "DoubleMLEstimator", thanks for the feedback!